consensus/src/quorum_store/types.rs

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_consensus_types::proof_of_store::LogicalTime;
+use aptos_crypto::HashValue;
+use aptos_types::transaction::SignedTransaction;
+use aptos_types::PeerId;
+use bcs::to_bytes;
+use serde::{Deserialize, Serialize};
+use std::mem;
+
+pub(crate) type BatchId = u64;
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq, Eq)]
+pub struct SerializedTransaction {
+    // pub(crate) for testing purposes
+    #[serde(with = "serde_bytes")]
+    pub(crate) bytes: Vec<u8>,
+}
+
+impl SerializedTransaction {
+    pub fn from_signed_txn(txn: &SignedTransaction) -> Self {
+        Self {
+            bytes: to_bytes(&txn).unwrap(),
+        }
+    }
+
+    pub fn len(&self) -> usize {
+        self.bytes.len()
+    }
+
+    pub fn bytes(&self) -> &Vec<u8> {
+        &self.bytes
+    }
+
+    pub fn take_bytes(&mut self) -> Vec<u8> {
+        mem::take(&mut self.bytes)
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct FragmentInfo {
+    epoch: u64,
+    batch_id: u64,
+    fragment_id: usize,
+    payload: Vec<SerializedTransaction>,
+    maybe_expiration: Option<LogicalTime>,
+}
+
+impl FragmentInfo {
+    fn new(
+        epoch: u64,
+        batch_id: u64,
+        fragment_id: usize,
+        fragment_payload: Vec<SerializedTransaction>,
+        maybe_expiration: Option<LogicalTime>,
+    ) -> Self {
+        Self {
+            epoch,
+            batch_id,
+            fragment_id,
+            payload: fragment_payload,
+            maybe_expiration,
+        }
+    }
+
+    pub(crate) fn take_transactions(self) -> Vec<SerializedTransaction> {
```

Comment:
this is typically called `into_transactions`, take mostly refers to a &mut self function